doc/api/cli.md

```diff
@@ -1086,6 +1086,16 @@ added: v22.3.0
 
 Enable [snapshot testing][] in the test runner.
 
+### `--experimental-typescript`
```

Comment:
Given this feature's goal is to run ts by default, I think we should have a `--no-experimental-typescript` instead and emit an experimental warning. Otherwise we will never catch any incompatibilities with `tsx`, `ts-node`, etc.

(We need the `--no-experimental-typescript` because of those likely incompatibilities).

Reply:
I would be extra cautious and ship it behind a flag at the beginning and then in a major, move it to `--no-experimental-typescript`

Reply:
`--no-experimental-typescript` is also added by this PR, but it's not documented because it's the default – that's how we usually document boolean flags, only the non-default one is documented.

IMO we should start by having the support opt-in; if your suggestion is to have it opt-out, I'm -1, we should land it opt-in and wait for bug reports first.